sktime/forecasting/mapa.py

```diff
@@ -0,0 +1,744 @@
+        X : pd.DataFrame or None, optional (default=None)
+            Exogenous variables for the forecasting model.
+        fh : ForecastingHorizon or None, optional (default=None)
+            Forecasting horizon for predictions
+
+        Returns
+        -------
+        self : MAPAForecaster
+            The fitted forecaster instance.
+        """
+        y = check_series(y)
+
+        self._y_cols = (
+            y.columns
+            if isinstance(y, pd.DataFrame)
+            else pd.Index([y.name])
+            if y.name
+            else pd.Index(["c0"])
+        )
+        self._y_name = y.name if isinstance(y, pd.Series) else None
+
+        y = self._ensure_positive_values(y)
+        y = self._handle_missing_data(y)
+
+        if isinstance(y, pd.Series):
+            y = pd.DataFrame(y)
+
+        valid_levels = []
+        for level in self._aggregation_levels:
+            try:
+                y_agg = self._aggregate(y, level)
+                y_agg.columns = self._y_cols
+
+                decomposed, seasonal_enabled, seasonal_period = self._decompose(
+                    y_agg, level
+                )
+
+                forecaster = type(self._base_forecaster)(
+                    **self._base_forecaster.get_params()
+                )
+
+                if not seasonal_enabled:
+                    forecaster_params = forecaster.get_params()
+                    if "seasonal" in forecaster_params:
+                        forecaster.set_params(seasonal=None)
+                else:
+                    forecaster_params = forecaster.get_params()
+                    if "seasonal" in forecaster_params:
+                        forecaster.set_params(seasonal="add", sp=seasonal_period)
+
+                trend_cols = [
+                    col for col in decomposed.columns if col.endswith("_trend")
+                ]
+                trend_data = decomposed[trend_cols].copy()
+                trend_data.columns = self._y_cols
+
+                forecaster.fit(trend_data)
+                self.forecasters[level] = forecaster
+                valid_levels.append(level)
+
+            except Exception as e:
+                warn(f"Failed to process level {level}: {str(e)}")
+                continue
+
+        if not valid_levels:
+            raise ValueError("Failed to fit any aggregation levels")
+
+        self._aggregation_levels = valid_levels
+        return self
+
+    def _predict(self, fh, X=None):
+        """Generate forecasts based on the fitted MAPA forecaster.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            Forecasting horizon defining the steps ahead to predict.
+        X : pd.DataFrame or None, optional (default=None)
+            Exogenous variables for prediction.
+
+        Returns
+        -------
+        pd.DataFrame
+            Forecasted values for the specified forecasting horizon.
+        """
+        fh = check_fh(fh)
```

Comment:
did not notice this before: you do not need to call the check, that happens in the boilerplate.